pkg/kapp/cmd/app/deploy.go

```diff
@@ -158,7 +159,7 @@ func (o *DeployOptions) Run() error {
 	}
 
 	existingResources, existingPodRs, err := o.existingResources(
-		newResources, labeledResources, resourceFilter, supportObjs.Apps, usedGKs, append(meta.LastChange.Namespaces, nsNames...))
+		newResources, labeledResources, resourceFilter, supportObjs.Apps, usedGKs, append(meta.LastChange.Namespaces, nsNames...), newApp)
```

Comment:
Can we explore based on whether it is newApp or not, if we make the call to `o.existingResources(...)`. Something like:
```
if !newApp {
existingResources, existingPodRs, err := o.existingResources(newResources, labeledResources, resourceFilter, supportObjs.Apps, usedGKs, append(meta.LastChange.Namespaces, nsNames...))
}
```
This way we will have minimal code change. Needs regression testing as well.

Reply:
We still need to check the existing non labeled resources even if it is a new app.
https://github.com/vmware-tanzu/carvel-kapp/blob/7e7b3a0485238f1e6c1f481835f102c12db95ac2/pkg/kapp/resources/labeled_resources.go#L119

Reply:
But I agree that we could probably make the check a little before then we have done now.
Maybe [here](https://github.com/vmware-tanzu/carvel-kapp/blob/7e7b3a0485238f1e6c1f481835f102c12db95ac2/pkg/kapp/resources/labeled_resources.go#L110)?

Reply:
I agree. Also, did we test the scenario: A resource already exists on cluster, and we try to kappify it. Also, can we have e2e test case around it if not already there?

Reply:
Does not seem like a test like that is there. Since we are touching those parts of the code, I think it makes sense to have a test which verifies using the diff summary that kapp adopts existing resources.

Reply:
added e2e test